src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/correspondence/intervener/IntervenerCorresponder.java

```diff
@@ -0,0 +1,140 @@
+        } else if (intervenerChangeDetails.getIntervenerType() == IntervenerType.INTERVENER_TWO) {
+            sendIntervenerCorrespondence(caseDetails.getData().getIntervenerTwoWrapper(), caseDetails, authToken);
+        } else if (intervenerChangeDetails.getIntervenerType() == IntervenerType.INTERVENER_THREE) {
+            sendIntervenerCorrespondence(caseDetails.getData().getIntervenerThreeWrapper(), caseDetails, authToken);
+        } else if (intervenerChangeDetails.getIntervenerType() == IntervenerType.INTERVENER_FOUR) {
+            sendIntervenerCorrespondence(caseDetails.getData().getIntervenerFourWrapper(), caseDetails, authToken);
+        }
+    }
+
+    @Override
+    protected void sendApplicantCorrespondence(FinremCaseDetails caseDetails, String auth) {
+        if (shouldSendApplicantSolicitorEmail(caseDetails)) {
+            log.info("Sending email correspondence to applicant for case: {}", caseDetails.getId());
+            this.emailApplicantSolicitor(caseDetails);
+        } else {
+            log.info("Sending letter correspondence to applicant for case: {}", caseDetails.getId());
+            String recipient = DocumentHelper.PaperNotificationRecipient.APPLICANT.toString();
+            bulkPrintService.sendDocumentForPrint(
+                getAppRepDocumentToPrint(caseDetails, auth,
+                    DocumentHelper.PaperNotificationRecipient.APPLICANT), caseDetails, recipient, auth);
+        }
+    }
+
+    @Override
+    protected void sendRespondentCorrespondence(FinremCaseDetails caseDetails, String auth) {
+        if (shouldSendRespondentSolicitorEmail(caseDetails)) {
+            log.info("Sending email correspondence to respondent for case: {}", caseDetails.getId());
+            this.emailRespondentSolicitor(caseDetails);
+        } else {
+            log.info("Sending letter correspondence to respondent for case: {}", caseDetails.getId());
+            String recipient = DocumentHelper.PaperNotificationRecipient.RESPONDENT.toString();
+            bulkPrintService.sendDocumentForPrint(
+                getAppRepDocumentToPrint(caseDetails, auth,
+                    DocumentHelper.PaperNotificationRecipient.RESPONDENT), caseDetails, recipient, auth);
+        }
+    }
+
+    protected void sendIntervenerCorrespondence(IntervenerWrapper intervenerWrapper, FinremCaseDetails caseDetails, String auth) {
+        if (shouldSendIntervenerSolicitorEmail(intervenerWrapper)) {
+            log.info("Sending email correspondence to {} for case: {}", intervenerWrapper.getIntervenerType(), caseDetails.getId());
+            String recipientName = intervenerWrapper.getIntervenerSolName();
+            String recipientEmail = intervenerWrapper.getIntervenerSolEmail();
+            String referenceNumber = intervenerWrapper.getIntervenerSolicitorReference();
+            notificationService.sendIntervenerSolicitorAddedEmail(caseDetails, intervenerWrapper,
```

Comment:
Why are we doing the added send here? this should be in the added corresponder considering we have a removed corresponder